.github/workflows/e2e-pw-pull-request.yml

```
@@ -0,0 +1,71 @@
+name: E2E Playwright Tests - Pull Request
+
+on:
+    pull_request:
+        branches:
+            - develop
+            - trunk
+            - try/playwright-e2e-tests
+    workflow_dispatch:
+    workflow_call:
+        inputs:
+            wcpay-use-build-artifact:
+                type: boolean
+                required: false
+                default: false
+            repo-branch:
+                type: string
+                required: false
+                description: 'Branch to be used for running tests'
+
+env:
+    E2E_GH_TOKEN: ${{ secrets.E2E_GH_TOKEN }}
+    WCP_DEV_TOOLS_REPO: ${{ secrets.WCP_DEV_TOOLS_REPO }}
+    WCP_DEV_TOOLS_BRANCH: 'trunk'
+    WCP_SERVER_REPO: ${{ secrets.WCP_SERVER_REPO }}
+    WC_SUBSCRIPTIONS_REPO: ${{ secrets.WC_SUBSCRIPTIONS_REPO }}
+    E2E_BLOG_ID: ${{ secrets.E2E_BLOG_ID }}
+    E2E_BLOG_TOKEN: ${{ secrets.E2E_BLOG_TOKEN }}
+    E2E_USER_TOKEN: ${{ secrets.E2E_USER_TOKEN }}
+    WC_E2E_SCREENSHOTS: 1
+    E2E_SLACK_CHANNEL: ${{ secrets.E2E_SLACK_CHANNEL }}
+    E2E_SLACK_TOKEN: ${{ secrets.E2E_SLACK_TOKEN }}
+    E2E_USE_LOCAL_SERVER: false
+    E2E_RESULT_FILEPATH: 'tests/e2e/results.json'
+    WCPAY_USE_BUILD_ARTIFACT: ${{ inputs.wcpay-use-build-artifact }}
+    WCPAY_ARTIFACT_DIRECTORY: 'zipfile'
+    NODE_ENV: 'test'
+    FORCE_E2E_DEPS_SETUP: true
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    wcpay-e2e-tests:
+        runs-on: ubuntu-latest
+
+        name: WC - latest
+
+        env:
+            E2E_WP_VERSION: 'latest'
+            E2E_WC_VERSION: 'latest'
+
+        steps:
+            - name: Checkout WCPay repository
+              uses: actions/checkout@v3
+              with:
+                  ref: ${{ inputs.repo-branch || github.ref }}
+
+            - name: 'Download WooCommerce Payments build file'
+              if: ${{ inputs.wcpay-use-build-artifact }}
+              uses: actions/download-artifact@v3
+              with:
+                  name: woocommerce-payments
+                  path: ${{ env.WCPAY_ARTIFACT_DIRECTORY }}
+
+            - name: Setup E2E environment
+              uses: ./.github/actions/e2e-pw/env-setup
```

```suggestion
              uses: ./.github/actions/e2e/env-setup
```

I think we can reuse the `./.github/actions/e2e/env-setup` composite action, and add the specific PW step for installing it right after, instead of creating a new env-setup action from scratch. 

```
- name: Install Playwright
          shell: bash
          run: |
              echo "::group::Install Playwright"
              npx playwright install
              echo "::endgroup::"
```

Or do you foresee any more specific setup to be done for PW, beside installing it (which should be done automatically when we add the playwright dependency in our package.json; why haven't we added playwright as dev dependency in that PR?)?

In any case, the `e2e-pw` composite action could run the `./.github/actions/e2e/env-setup` and add specific steps after, if we really want to encapsulate the whole PW setup thing in a different file

---

> why haven't we added playwright as dev dependency in that PR?

`@playwright/test` is installed as a dev dependency, but it seems that Playwright needs to run an additional browser installation step with `npx playwright install` ([see docs](https://playwright.dev/docs/ci-intro#on-pushpull_request)).

https://github.com/Automattic/woocommerce-payments/blob/d98d70548b8f536e43fee787db370415939c04f0/package.json#L87

---

> we can reuse the `./.github/actions/e2e/env-setup` composite action

I've made this change in 83c1e6b and added the specific Playwright install step.